ext/liquid_c/lexer.c

```diff
@@ -109,7 +109,7 @@ const char *lex_one(const char *start, const char *end, lexer_token_t *token)
         }
     }
 
-    if (ISDIGIT(c) || c == '-') {
+    if (ISDIGIT(c) || (c == '-' && str + 1 < end && ISDIGIT(str[1]))) {
```

Comment:
This will make the strict parser more strict, so could be a breaking change.  We should instead use `Liquid::Usage.increment` to instrument this change to determine if it will break any code in practice.  If we determine that it is relied upon, then I think we will need to make the corresponding strict parsing in the ruby liquid gem less strict instead.

Reply:
I opened #123 to instrument this usage.